core/util/src/mill/util/Jvm.scala

```diff
@@ -620,12 +620,19 @@ object Jvm {
       id: String,
       ctx: Option[mill.define.TaskCtx] = None,
       coursierCacheCustomizer: Option[FileCache[Task] => FileCache[Task]] = None,
-      jvmIndexVersion: String = mill.api.BuildInfo.coursierJvmIndexVersion
+      jvmIndexVersion: String = mill.api.BuildInfo.coursierJvmIndexVersion,
+      useShortPaths: Boolean = false
   ): Result[os.Path] = {
     val coursierCache0 = coursierCache(ctx, coursierCacheCustomizer)
+    val shortPathDirOpt = Option.when(useShortPaths) {
+      if (isWin) os.Path(System.getenv("UserProfile")) / ".mill/cache/jvm"
+      else os.home / ".cache/mill/jvm"
```

Comment:
This should respect the `XDG_CACHE_HOME` env var. We respect it in other places like the bootstrap scripts and I think coursier also repects it.

Reply:
Should be fixed